docs/DeveloperGuide.adoc

```diff
@@ -763,20 +848,197 @@ Use case ends.
+=== Use case: UC5 - List contacts after applying filtered search
+
+*MSS*
+
+1.  User specifies the keywords for different link:#attributes[attributes] and requests the list
+2.  iContacts finds list of all people satisfying all attributes
+3.  iContacts displays list of all such contacts
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty
++
+Use case ends.
+
+[discrete]
+=== Use case: UC6 - Get directions to a contact's address
+
+*MSS*
+
+1.  User requests for the list of contacts
+2.  iContacts displays list of all such contacts
+3.  User requests for directions to a contact's address
+4.  iContacts shows directions from current location to contact's address in Google Maps using link:#GUI[GUI]
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. iContacts shows an error message.
++
+Use case resumes at step 2.
+
+[none]
+* 4a. Google Maps unable to find location of direction to address
++
+[none]
+** 4a1. iContacts shows an error message through GUI
++
+Use case ends.
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any link:#mainstream-os[mainstream OS] as long as it has Java `1.8.0_60` or higher installed.
 .  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-{More to be added}
+.  A new user to a link:#CLI[CLI] environment and/or address book application should be able to learn how to install and use the application within 20 minutes with the help of the User Guide.
+.  Should be responsive and commands are executed within 2 seconds under normal load (1000 persons or less) and typical usage.
+.  Essential services (adding, deleting contacts etc) should still work perfectly without Internet access.
+.  Layout of the application should be clear, appealing and attractive.
+.  Should be able to handle any invalid input and recover from it.
+.  Should not consume too much processing time/power, less than 10% of the CPU of modern computers/laptops.
+.  Should take up as little memory on the hard disk/permanent memory as possible, less than 100 MB.
+.  Should be free to download and use.
+.  Should work on 32-bit and 64-bit environments.
+.  Should not display content not associated with the operations of the address book.
+.  User Guide and Developer Guide should be well-documented.
+.  Should be backwards compatible with storage files from prior versions.
+.  Should be compatible with storage files from the same application but from different computers.
+.  Should minimize dependencies on third-party software.
+.  Should consistently come up with at least one new feature every month.
+.  Should require some form of authentication before allowing usage.
```

Comment:
i guess the ones below are all related to process or design. not really NFR

Reply:
Agreed. Removed the ones which aren't NFR's